components/async-operation-status-mgt/org.wso2.carbon.identity.framework.async.operation.status.mgt/src/main/java/org/wso2/carbon/identity/framework/async/operation/status/mgt/api/buffer/SubOperationStatusObject.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2025, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.framework.async.operation.status.mgt.api.buffer;
+
+import org.wso2.carbon.identity.framework.async.operation.status.mgt.api.constants.OperationStatus;
+
+/**
+ * Represents the status of a single sub-operation within an asynchronous operation.
+ * This object holds a status value (e.g., "SUCCESS" or "FAIL") that can be used
+ * to evaluate the overall operation's result.
+ */
+public class SubOperationStatusObject {
```

Comment:
Can this class be replaced by the `OperationStatus` enum?. if so shall we remove this class